### ISSUE TITLE
SABnzbd health check is too eager

### ISSUE THREAD

**Issue by fschoenm** [Sonarr/Sonarr#6044]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

Sonarr complains that "Download client SABnzbd is set to remove completed downloads. This can result in downloads being removed from your client before Sonarr can import them.". According to the code [here](https://github.com/Sonarr/Sonarr/blame/c7824bb593291634bf14a5f7aa689666969b03bf/src/NzbDrone.Core/Download/Clients/Sabnzbd/Sabnzbd.cs#L279), the check fails if history retention is not disabled (set to 0).

My history_retention is set to 31 days which seems reasonable to me. Sonarr may not be the only client being served by SABnzbd and I think I need to configure it to not keep everything forever just so that Sonarr does not complain. 31 days should be enough time to import anything.

The change in question was added in #6020.

### Expected Behavior

Don't complain about reasonable SABnzbd configuration. Maybe the check should be against "-1" which I think means that everything is removed immediately (see https://github.com/sabnzbd/sabnzbd/blob/1e4ef9c38123abfe3701382f9d4646d288f6d52f/sabnzbd/database.py#L247).

### Steps To Reproduce

1. Use SABnzbd.
2. Set history retention to 31 days (Switches -> Post processing)
3. Check Sonarr status messages.

### Environment

```markdown
- OS: Ubuntu 20.04
- Sonarr: 4.0.0.682
- Docker Install: yes
- Using Reverse Proxy: yes
- Browser: 
- Database:
```

### What branch are you running?

Develop

### Trace Logs?

n/a

**Comment by thezoggy** [comment]
the options are:
![image](https://github.com/Sonarr/Sonarr/assets/500882/920fa265-f8be-4e2c-b45b-9da70b2b828f)

ideally the health check should prob mainly try and catch those that use "do not keep any completed jobs" or those that use keep x jobs (since x jobs can easily be exhausted if they use a low value or have multiple apps since x would be reached).

I do agree that people doing keep jobs for x days is prob safe, assuming jobs dont take days to be processed..  so more than 30 days is prob just fine as if its taking 30 days to work down their completed jobs they have much bigger issues.

**Comment by stevietv** [comment]
I'd propose to switch the check to this:
```
if (config.Misc.history_retention.slice(-1) == "d")
{
    status.RemovesCompletedDownloads = config.Misc.history_retention.slice(0,config.Misc.history_retention.length-1) < 7;
} else
{
    status.RemovesCompletedDownloads = config.Misc.history_retention!= "0";
}
```

Catches anyone who has it set to less than 7days, or set to do not keep, or set to a number of jobs.

**Comment by abs3ntdev** [comment]
Experiencing similar behavior on qbittorrent. I have my retention set to 6 weeks but am still receiving the warning. If the fix could also be applied to qbittorrent that would be appreciated.

**Comment by stevietv** [comment]
> Experiencing similar behavior on qbittorrent. I have my retention set to 6 weeks but am still receiving the warning. If the fix could also be applied to qbittorrent that would be appreciated.
> 

The check in Qbittorrent is because your instance is set to remove torrents instead of pause torrents. When using Sonarr with Qbittorrent you should be setting Qbit to pause on reaching the seeding goals and enable remove finished downloads in sonarrs qbit settings

**Comment by stevietv** [comment]
@abs3ntdev PR #6050 will remove the warning if seeding time is more than 14 days

**Comment by abs3ntdev** [comment]
> @abs3ntdev PR #6050 will remove the warning if seeding time is more than 14 days

Thank you! I appreciate it

**Comment by justinbhopper** [comment]
@stevietv Would it be possible to update the error message so that users know about the new "at least 14 days" rule?  Currently the error message still implies that removing _any_ history is a bad thing.

**Comment by stevietv** [comment]
> @stevietv Would it be possible to update the error message so that users know about the new "at least 14 days" rule? Currently the error message still implies that removing _any_ history is a bad thing.

I've updated the wiki article slightly that this health check links to when active. This should be sufficient? https://wiki.servarr.com/en/sonarr/system#download-client-removes-completed-downloads

**Comment by justinbhopper** [comment]
@stevietv That's perfect, thank you!
